### ISSUE TITLE
cache doesn't carry over between runs

### ISSUE THREAD

**Issue by kousu** [santropolroulant/smell-test#7]
```
[kousu@requiem smell-test]$ sudo ./smell-test 
[sudo] password for kousu: 
[**] Beginning "Smell Test"



[+] User asked for "google.com." DNS responded "172.217.3.174"
[+] Evaluating google.com

 Start 2018-03-08 15:35:49        -->> 216.58.216.142:443 (google.com) <<--
...

$ ls results/google*
google.com_20180308-152350.json

[kousu@requiem smell-test]$ sudo ./smell-test 
[sudo] password for kousu: 
[**] Beginning "Smell Test"



[+] User asked for "google.com." DNS responded "172.217.3.174"
[+] Evaluating google.com

 Start 2018-03-08 15:35:49        -->> 216.58.216.142:443 (google.com) <<--
...
Ctrl-C
```

The cache should check the results/ folder to know if it needs to re-query, I think. I guess right now the cache is being held in a python dict only?

**Comment by johnsaigle** [comment]
> python dict only
yep! definitely needs real caching

**Comment by johnsaigle** [comment]
This is solved as of #21
